File: commons/test-utils/src/test/kotlin/com/cedarsoft/test/utils/JsonUtilsTest.java
package com.cedarsoft.test.utils;

import static org.assertj.core.api.AssertionsForClassTypes.*;

import org.junit.*;
import org.junit.jupiter.api.Test;

/**
 */
class JsonUtilsTest {
  @Test
  void testIt() throws Exception {
    try {
      JsonUtils.assertJsonEquals("[]", "[\"asdf\"]");
      fail("Where is the Exception");
    }
    catch (ComparisonFailure e) {
      assertThat(e).hasMessage("JSON comparison failed expected:<[ []]> but was:<[ [\"asdf\" ]]>");
    }
  }
}

File: commons/test-utils/src/test/kotlin/it/neckar/open/test/utils/JsonUtilsTest.kt
package it.neckar.open.test.utils

import it.neckar.open.test.utils.JsonUtils.assertJsonEquals
import org.assertj.core.api.AssertionsForClassTypes
import org.junit.ComparisonFailure
import org.junit.jupiter.api.Test

/**
 */
class JsonUtilsTest {
  @Test
  fun testIt() {
    try {
      assertJsonEquals("[]", "[\"asdf\"]")
      AssertionsForClassTypes.fail("Where is the Exception")
    } catch (e: ComparisonFailure) {
      AssertionsForClassTypes.assertThat(e).hasMessage("JSON comparison failed expected:<[ []]> but was:<[ [\"asdf\" ]]>")
    }
  }
}
